src/session/client-certificate.c

```diff
@@ -52,15 +67,15 @@
  * it.
  */
 static char *
-read_proc_self_cgroup (size_t *out_length)
+read_proc_pid_cgroup (int dirfd, size_t *out_length)
 {
-  FILE *fp = fopen ("/proc/self/cgroup", "r");
-
-  if (fp == NULL)
+  int fd = openat (dirfd, "cgroup", O_RDONLY);
+  if (fd < 0)
     {
-      warn ("Failed to open /proc/self/cgroup");
+      warn ("Failed to open /proc/pid/cgroup");
       return NULL;
     }
+  FILE *fp = fdopen (fd, "r");
 
   /* Support cgroups v1 by looping.
```

Comment:
Do we still need cgroups v1 support?  Might be a nice cleanup.

Reply:
I checked Debian stable, RHEL 9.6, and Ubuntu 22.04. Sent #21260, blocking on this as we can then clean this up further.